rclcpp/test/rclcpp/test_publisher.cpp

```diff
@@ -413,10 +415,18 @@ TEST_F(TestPublisher, intra_process_publish_failures) {
 
   {
     rclcpp::LoanedMessage<test_msgs::msg::Empty> loaned_msg(*publisher, allocator);
-    loaned_msg.release();
+    auto msg = loaned_msg.release();  // this will unmanage the ownership of the message
     RCLCPP_EXPECT_THROW_EQ(
       publisher->publish(std::move(loaned_msg)),
       std::runtime_error("loaned message is not valid"));
+    // if the message is actually loaned from the middleware but not be published,
+    // it is user responsibility to return the message to the middleware manually
+    if (publisher->can_loan_messages()) {
```

Comment:
if this is not issued, rmw_fastrtps Fast-DDS will fail to destroy DDS writer since the allocated memory in Fast-DDS has not returned yet.